amaranth/lib/io.py

```diff
@@ -253,6 +258,362 @@ def __repr__(self):
         return f"DifferentialPort({self._p!r}, {self._n!r}, invert={invert!r}, direction={self._direction})"
 
 
+class Buffer(wiring.Component):
+    """A combinatorial I/O buffer.
+
+    Parameters
+    ----------
+    direction : Direction
+    port : SingleEndedPort, DifferentialPort, or a custom platform-specific type
```

Comment:
Needs a comment explaining when it can be a custom platform-specific type. (I think you can specify a parameter twice and create a conditional this way?)